subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/ToBeFixedForInstantExecutionExtension.groovy

```diff
@@ -17,207 +17,181 @@
-        return bottomSpecs.length == 0 || bottomSpecs.any { specNamePredicate.test(it) }
-    }
-
-    static boolean iterationMatches(String[] iterationMatchers, String iterationName) {
-        return isAllIterations(iterationMatchers) || iterationMatchers.any { iterationName.matches(it) }
-    }
-
-    static boolean isAllIterations(String[] iterationMatchers) {
-        return iterationMatchers.length == 0
-    }
-
     @Override
     void visitFeatureAnnotation(ToBeFixedForInstantExecution annotation, FeatureInfo feature) {
-        // This override is required to satisfy spock's zealous runtime checks
-    }
-
-    /**
-     * Spock spec listener that registers failure collecting method interceptors for the given feature.
-     *
-     * Registration happens as late as possible and to the utmost position in the stack in order to catch
-     * failures without being sensible to other spock extensions that re-order interceptors nor to the place
-     * of annotated spock features in hierarchies of spec classes.
-     *
-     * This is also so that failures in setup and cleanup methods are taken into account by the annotation.
-     * Unfortunately this leads to not being able to throw from inside the interceptors but instead throw
-     * from outside the test execution, after setup/test/cleanup, in Spock's core, effectively stopping the
-     * whole spec execution. In practice, as soon as a test doesn't behave, we stop executing subsequent tests.
-     */
-    private static class CatchFeatureFailuresRunListener extends AbstractRunListener {
-
-        private final FeatureInfo feature
-        private final String[] iterationMatchers
-        private final RecordFailuresInterceptor failuresInterceptor
-        private final FeatureFilterInterceptor fixturesInterceptor
 
-        CatchFeatureFailuresRunListener(FeatureInfo feature, String[] iterationMatchers) {
-            this.feature = feature
-            this.iterationMatchers = iterationMatchers
-            this.failuresInterceptor = new RecordFailuresInterceptor(iterationMatchers)
-            this.fixturesInterceptor = new FeatureFilterInterceptor(feature, iterationMatchers, failuresInterceptor)
+        if (GradleContextualExecuter.isNotInstant()) {
+            return
         }
 
-        @Override
-        void beforeSpec(SpecInfo spec) {
-            spec.specsTopToBottom*.each { s ->
-                s.setupMethods*.interceptors*.add(0, fixturesInterceptor)
-                s.cleanupMethods*.interceptors*.add(0, fixturesInterceptor)
-            }
+        if (!isEnabledSpec(annotation, feature)) {
+            return
         }
 
-        @Override
-        void beforeFeature(FeatureInfo featureInfo) {
-            if (featureInfo == feature) {
-                featureInfo.featureMethod.interceptors.add(0, failuresInterceptor)
-            }
+        if (annotation.skip() == FLAKY) {
```

Comment:
:question: Shouldn't this be `annotation.skip() != DO_NOT_SKIP` instead?

Reply:
Yeah, it was causing quite a bunch of CI failures because of not cleaned up test files.
Fix in https://github.com/gradle/gradle/pull/13646

Reply:
I see. Thank you!